### PR TITLE
feat: filter out non-native deps from config output; fix missing hooks

### PR DESCRIPTION
**Pull request by thymikee**
Summary:
---------

This diff aims to filter out unnecessary non-native dependencies from the `config` command output and display dependency linking "hooks" properly (needed for https://github.com/react-native-community/cli/issues/429), which were not there because we stored them as non-serializable functions.

Before:
```sh
$ yarn react-native config
{
  "dependencies": {
    "dep-1": { ... },
    "dep-2": { ... },
    "native-dep-1": { 
   	  "hooks": {}
    },
  }
  ...
}
```

After:
```sh
$ yarn react-native config
{
  "dependencies": {
    "native-dep-1": { 
   	  "hooks": {
		"postlink": "node script.js"
      }
    },
  }
  ...
}
```

Test Plan:
----------

Updated tests accordingly.

### CODE REVIEW

File: packages/cli/src/tools/config/index.js [hunk 1]
```diff
@@ -2,36 +2,90 @@
...
+  );
+}
 
 /**
  * Loads CLI configuration
  */
 function loadConfig(projectRoot: string = process.cwd()): ConfigT {
   const userConfig = readConfigFromDisk(projectRoot);
 
+  const initialConfig: ConfigT = {
```
Comment (thymikee) [inline packages/cli/src/tools/config/index.js]: this block was moved without modifications

File: packages/cli/src/tools/config/index.js [hunk 2]
```diff
@@ -2,36 +2,90 @@
...
+            : finalConfig.platforms[platform].dependencyConfig(
+                root,
+                config.dependency.platforms[platform] || {},
+              );
+          return dependency;
+        },
+        {},
+      ),
+      assets: findAssets(root, config.dependency.assets),
+      hooks: config.dependency.hooks,
```
Comment (thymikee) [inline packages/cli/src/tools/config/index.js]: the only change in this moved block is this line – we pass the `hooks` string directly, without mapping to functions (which are not serializable, hence not displayed after JSON.stringify runs on config)

File: packages/cli/src/commands/config/config.js [hunk 3]
```diff
@@ -2,10 +2,31 @@
...
+  });
+  return filtered;
+}
+
 export default {
   name: 'config',
   description: 'Print CLI configuration',
   func: async (argv: string[], ctx: ConfigT) => {
-    console.log(JSON.stringify(ctx, null, 2));
+    console.log(JSON.stringify(filterConfig(ctx), null, 2));
```
Comment (grabbou) [inline packages/cli/src/commands/config/config.js]: This is a good idea to do it here - we still keep the lazy behaviour in other places

File: packages/cli/src/tools/config/index.js (hunk 2 above)
Comment (grabbou) [reply packages/cli/src/tools/config/index.js]: Yeah, right!

Comment (grabbou) [reply packages/cli/src/tools/config/index.js]: I guess an alternative take would be to provide a special `toString` implementation as a result of `makeHook` call to make it serializable?

Comment (thymikee) [reply packages/cli/src/tools/config/index.js]: I think it's better to keep the whole config serializable anyway, less things to remember

File: packages/cli/src/commands/config/config.js (hunk 3 above)
Comment (thymikee) [reply packages/cli/src/commands/config/config.js]: Yup, that was the intention. Should I leave a comment to make it obvious?

Comment (grabbou) [reply packages/cli/src/commands/config/config.js]: I think it's fine like this :)

File: packages/cli/src/tools/config/index.js (hunk 2 above)
Comment (grabbou) [reply packages/cli/src/tools/config/index.js]: Yeah, I was suggesting to make `makeHook` return a serializable function (`toString` returns a path). 

They are going to be gone (most likely) in the future anyway, so not bothering to much with that one.

File: packages/cli/src/commands/config/config.js (hunk 3 above)
Comment (thymikee) [reply packages/cli/src/commands/config/config.js]: Feel free to approve it then? 🙂

Comment (grabbou) [reply packages/cli/src/commands/config/config.js]: Be my guest

**Pull request merged**